File: Python/add-two-numbers.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        p = l1
        num = 0
        count = 0
        while p:
            num += (10**count) * p.val
            count += 1
            p = p.next

        p = l2
        count = 0
        while p:
            num += (10**count) * p.val
            count += 1
            p = p.next

        l = []
        if num == 0:
            l.append(0)
        else:
            while num > 0:
                l.append(num % 10)
                num = num // 10

        start = None
        count = 0
        for n in l:
            if count == 0:
                start = ListNode(n)
                p = start
                count += 1
            else:
                p.next = ListNode(n)
                p = p.next
                count += 1

        return start

File: Java/add-two-numbers.java
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        // 新建一个链表用来存储结果
        ListNode ans = new ListNode(0), p = ans;

        // 进位信息
        int carry = 0;
        int num1, num2, temp;
        // 开始遍历
        while (l1 != null || l2 != null) {
            num1 = l1 == null ? 0 : l1.val;
            num2 = l2 == null ? 0 : l2.val;
            temp = num1 + num2 + carry;
            carry = temp > 9 ? 1 : 0;
            temp = temp > 9 ? temp - 10 : temp;
            p.next = new ListNode(temp);
            p = p.next;
            l1 = l1 == null ? null : l1.next;
            l2 = l2 == null ? null : l2.next;
        }

        if (carry == 1) {
            p.next = new ListNode(1);
        }

        return ans.next;
    }
}
